core/src/main/scala/org/broadinstitute/dsde/rawls/entities/compact/CompactEntityProvider.scala

```diff
@@ -54,7 +74,43 @@ class CompactEntityProvider(implicit protected val executionContext: ExecutionCo
                             parentContext: RawlsRequestContext
   ): Future[EntityCopyResponse] = ???
 
-  override def createEntity(entity: Entity, parentContext: RawlsRequestContext): Future[Entity] = ???
+  override def createEntity(entity: Entity, parentContext: RawlsRequestContext): Future[Entity] = {
+    EntityUtils.validateEntity(entity)
+    val createFuture = repository.dataSource.inTransaction { _ =>
+      for {
+        // does this entity already exist?
+        preExisting <- repository.queries.getEntity(workspaceId, entity.entityType, entity.name)
+        _ = if (preExisting.nonEmpty)
+          throw new RawlsExceptionWithErrorReport(
+            errorReport = ErrorReport(
+              StatusCodes.Conflict,
+              s"${entity.entityType} ${entity.name} already exists in ${requestArguments.workspace.toWorkspaceName}"
+            )
+          )
+        // find all references in this entity
+        refs: Map[AttributeName, Seq[AttributeEntityReference]] = findAllReferences(entity)
+        // find all unique references in this entity
+        uniqueRefs: Set[AttributeEntityReference] = refs.values.flatten.toSet
+        // verify that all references in the entity-to-be-saved actually exist
+        referencedIds <- repository.queries.getReferencedIds(workspaceId, uniqueRefs)
+        _ = if (uniqueRefs.size != referencedIds.size)
+          throw new EntityReferenceNotFoundException("Some entity references do not exist")
+        // save the entity
+        _ <- repository.queries.createEntity(workspaceId, entity)
+        // did it save correctly? re-retrieve it. By re-retrieving it, we can 1) get its id, and 2) get the actual,
+        // normalized JSON that was persisted to the db. When we return the entity to the user, we return the
+        // normalized version.
+        savedEntityRecordOption <- repository.queries.getEntity(workspaceId, entity.entityType, entity.name)
+        savedEntityRecord = savedEntityRecordOption.getOrElse(throw new DataEntityException("Could not save entity"))
+        // save all references from this entity to other entities
+        _ <- replaceReferences(savedEntityRecord.id, referencedIds.toSet, isInsert = true)
+      } yield savedEntityRecord.toEntity
+    }
+
+    withWorkspaceLastModified(createFuture)
```

Comment:
nit: It might be worth adding an explanatory comment here to reinforce the fact that this is intentionally called in this way since it is unusual to drop the `Future` like this